kyo-core/shared/src/main/scala/kyo/Stream.scala

```diff
@@ -0,0 +1,132 @@
+package kyo
+
+extension (streamObj: Stream.type)
+    /** Merges multiple streams asynchronously. Stream stops when all sources streams have completed.
+      *
+      * @note
+      *   Merges chunks and not individual elements. Rechunk source streams to size 1 to interleave individual elements.
+      * @note
+      *   Resulting stream does not preserve chunking from sources
+      * @param streams
+      *   Sequence of streams to be merged
+      * @param bufferSize
+      *   Size of the buffer that source streams will write to and outputs stream will read from
+      */
+    inline def mergeAll[V, S](streams: Seq[Stream[V, S]], bufferSize: Int = 1024)(using Frame): Stream[V, S & Async] =
+        Stream:
+            Resource.run:
+                for
+                    channel <- Resource.acquireRelease(Channel.init[Option[V]](bufferSize, Access.MultiProducerMultiConsumer))(_.close)
+                    _ <- Async.run(Abort.run(
+                        Async
+                            .foreachDiscard(streams)(
+                                _.foreachChunk(v => channel.putBatch(v.map(Some(_))))
+                            )
+                            .andThen(channel.put(None))
+                    ))
+                    _ <- channel
+                        .streamUntilClosed()
+                        .collectWhile(Maybe.fromOption)
+                        .emit
+                yield ()
+
+    /** Merges multiple streams asynchronously. Stream stops as soon as any of the source streams complete.
+      *
+      * @note
+      *   Merges chunks and not individual elements. Rechunk source streams to size 1 to interleave individual elements.
+      * @note
+      *   Resulting stream does not preserve chunking from sources
+      * @param streams
+      *   Sequence of streams to be merged
+      * @param bufferSize
+      *   Size of the buffer that source streams will write to and outputs stream will read from
+      */
+    inline def mergeAllHalting[V, S](streams: Seq[Stream[V, S]], bufferSize: Int = 1024)(using Frame): Stream[V, S & Async] =
+        Stream:
+            Resource.run:
+                for
+                    channel <- Resource.acquireRelease(Channel.init[Option[V]](bufferSize, Access.MultiProducerMultiConsumer))(_.close)
```

Comment:
Why not `Maybe[V]`?

Reply:
I can't stream `Maybe` since there is no tag for it

Reply:
Now that I'm passing `Chunk` directly, I'm just using empty chunks to signal end-of-stream (empty chunks are not passed to the `Channel` from source streams)